web/src/components/organisms/Settings/Members/index.tsx

```diff
@@ -0,0 +1,252 @@
+  },
+  {
+    title: "Thumbnail",
+    dataIndex: "thumbnail",
+    key: "thumbnail",
+  },
+  {
+    title: "Email",
+    dataIndex: "email",
+    key: "email",
+  },
+  {
+    title: "Role",
+    dataIndex: "role",
+    key: "role",
+  },
+  {
+    title: "Action",
+    dataIndex: "action",
+    key: "action",
+  },
+];
+
+const Members: React.FC = () => {
+  const { workspaceId } = useParams();
+  const [collapsed, setCollapsed] = useState(false);
+
+  const { confirm } = Modal;
+
+  const {
+    user,
+    personalWorkspace,
+    workspaces,
+    handleModalClose,
+    handleModalOpen,
+    modalShown,
+    handleWorkspaceCreate,
+  } = useDashboardHooks(workspaceId);
+
+  const {
+    me,
+    currentWorkspace,
+    searchedUser,
+    changeSearchedUser,
+    handleUserSearch,
+    handleMemberAddToWorkspace,
+    handleMemberOfWorkspaceUpdate,
+    selectedMember,
+    roleModalShown,
+    handleMemberRemoveFromWorkspace,
+    handleRoleModalClose,
+    handleRoleModalOpen,
+    handleMemberCreationModalClose,
+    handleMemberCreationModalOpen,
+    memberCreationModalShown,
+  } = useHooks({ workspaceId });
+
+  const members = currentWorkspace?.members;
+
+  const showConfirm = useCallback(
+    (member: any) => {
+      confirm({
+        title: "Are you sure to remove this member?",
+        icon: <ExclamationCircleOutlined />,
+        content:
+          "Remove this member from workspace means this member will not view any content of this workspace.",
+        onOk() {
+          handleMemberRemoveFromWorkspace(member?.userId);
+        },
+        onCancel() {},
+      });
+    },
+    [confirm, handleMemberRemoveFromWorkspace]
+  );
+
+  const dataSource = members?.map((member) => ({
+    key: member.userId,
+    name: member.user.name,
+    thumbnail: (
+      <Avatar style={{ color: "#fff", backgroundColor: "#3F3D45" }}>
+        {member.user.name.charAt(0)}
+      </Avatar>
+    ),
+    email: member.user.email,
+    role: member.role,
+    action: (
+      <>
+        {member.userId !== me?.id && (
+          <a onClick={() => handleRoleChange(member)}>Change Role</a>
+        )}
+        {member.role !== "OWNER" && (
+          <a
+            style={{ marginLeft: "8px" }}
+            onClick={() => handleMemberDelete(member)}
+          >
+            Remove
+          </a>
+        )}
+      </>
+    ),
+  }));
+
+  const handleRoleChange = useCallback(
+    (member: any) => {
+      handleRoleModalOpen(member);
+    },
+    [handleRoleModalOpen]
+  );
+
+  const handleMemberDelete = useCallback(
+    (member: any) => {
+      showConfirm(member);
+    },
+    [showConfirm]
+  );
+
+  const handleMemberAdd = useCallback(() => {
+    if (!searchedUser) return;
+    handleMemberAddToWorkspace([searchedUser.id]);
+    changeSearchedUser(undefined);
+    handleMemberCreationModalClose();
+  }, [
+    searchedUser,
+    handleMemberAddToWorkspace,
+    changeSearchedUser,
+    handleMemberCreationModalClose,
+  ]);
```

Comment:
Can any of this be 1) Moved to the molecule components? Or 2) Be moved into the hooks file?
Especially functions like
```
const handleRoleChange = useCallback(
    (member: any) => {
      handleRoleModalOpen(member);
    },
    [handleRoleModalOpen]
  );
```
I feel like they aren't needed. You aren't doing anything new that the original `handleRoleModalOpen` isn't doing.